smtp.go

```diff
@@ -52,5 +66,29 @@ type MailOptions struct {
 	Auth *string
 }
 
+type DSNNotify string
+
+const (
+	NotifyNever   DSNNotify = "NEVER"
+	NotifyDelayed DSNNotify = "DELAY"
+	NotifyFailure DSNNotify = "FAILURE"
+	NotifySuccess DSNNotify = "SUCCESS"
+)
+
+type DSNAddressType string
+
+const (
+	AddressTypeRFC822 DSNAddressType = "RFC822"
+	AddressTypeUTF8   DSNAddressType = "UTF-8"
+)
```

Comment:
It seems a bit inconsistent to have the types prefixed with "DSN" but not the constants. Is this deliberate? Should we prefix everything with "DSN", or do you think we can drop the "DSN" prefix from type names?

(Sorry for the delay!)

Reply:
These follow the naming conventions adopted by @foxcpp in #135. I have no particular preference.

Reply:
I've went with the prefix everywhere.